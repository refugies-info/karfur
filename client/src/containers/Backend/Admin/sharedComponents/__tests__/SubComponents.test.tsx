// @ts-nocheck
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";
import {
  TypeContenu,
  Title,
  Structure,
  StyledStatus,
  ValidateButton,
  SeeButton,
  DeleteButton,
  TabHeader,
  FilterButton,
} from "../SubComponents";

let component;

test("should render TypeContenu with dispositif, detailed vue", () => {
  component = TestRenderer.create(
    <TypeContenu type={"dispositif"} isDetailedVue={true} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TypeContenu with demarche, detailed vue", () => {
  component = TestRenderer.create(
    <TypeContenu type={"demarche"} isDetailedVue={true} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TypeContenu with dispositif, not detailed vue", () => {
  component = TestRenderer.create(
    <TypeContenu type={"dispositif"} isDetailedVue={false} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TypeContenu with demarche, not detailed vue", () => {
  component = TestRenderer.create(
    <TypeContenu type={"demarche"} isDetailedVue={false} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Title with titre marque", () => {
  component = TestRenderer.create(
    <Title titreInformatif="titre informatif" titreMarque="titre marque" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Title without titre marque", () => {
  component = TestRenderer.create(<Title titreInformatif="titre informatif" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with null sponsor", () => {
  component = TestRenderer.create(<Structure sponsor={null} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor without nom", () => {
  const sponsor = { name: "test" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor actif", () => {
  const sponsor = { _id: "abc", nom: "test", status: "Actif" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor en attente", () => {
  const sponsor = { _id: "abc", nom: "test", status: "En attente" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render Structure with sponsor supprime", () => {
  const sponsor = { _id: "abc", nom: "test", status: "Supprimé" };
  component = TestRenderer.create(<Structure sponsor={sponsor} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus actif ", () => {
  component = TestRenderer.create(<StyledStatus text="Actif" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus en attente ", () => {
  component = TestRenderer.create(<StyledStatus text="En attente" />);
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Brouillon ", () => {
  component = TestRenderer.create(<StyledStatus text="Brouillon" />);
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus En attente non prioritaire ", () => {
  component = TestRenderer.create(
    <StyledStatus text="En attente non prioritaire" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Rejeté structure ", () => {
  component = TestRenderer.create(<StyledStatus text="Rejeté structure" />);
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus En attente admin ", () => {
  component = TestRenderer.create(<StyledStatus text="En attente admin" />);
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Accepté structure ", () => {
  component = TestRenderer.create(<StyledStatus text="Accepté structure" />);
  expect(component.toJSON()).toMatchSnapshot();
});
test("should render StyledStatus Supprimé ", () => {
  component = TestRenderer.create(<StyledStatus text="Supprimé" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus no corresponding status ", () => {
  component = TestRenderer.create(<StyledStatus text="test" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus no corresponding with override ", () => {
  component = TestRenderer.create(
    <StyledStatus text="Actif" overrideColor={true} textToDisplay="test" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render StyledStatus no corresponding with color ", () => {
  component = TestRenderer.create(
    <StyledStatus
      text="Actif"
      overrideColor={false}
      textToDisplay="test"
      textColor="textColor"
      color="color"
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render ValidateButton disabled", () => {
  component = TestRenderer.create(
    <ValidateButton onClick={() => {}} disabled={true} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render ValidateButton not disabled", () => {
  component = TestRenderer.create(
    <ValidateButton onClick={() => {}} disabled={false} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should call onClick ValidateButton", () => {
  const onClick = jest.fn();
  component = TestRenderer.create(
    <ValidateButton onClick={onClick} disabled={false} />
  );
  component.root.findByProps({ "data-test-id": "validate-button" }).props.onClick();
  expect(onClick).toHaveBeenCalled();
});

test("should render SeeButton", () => {
  component = TestRenderer.create(<SeeButton burl="test-burl" />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render DeleteButton disabled", () => {
  component = TestRenderer.create(
    <DeleteButton disabled={true} onClick={() => {}} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render DeleteButton not disabled", () => {
  component = TestRenderer.create(
    <DeleteButton disabled={false} onClick={() => {}} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should call onClick DeleteButton", () => {
  const onClick = jest.fn();
  component = TestRenderer.create(
    <DeleteButton onClick={onClick} disabled={false} />
  );
  component.root.findByProps({ "data-test-id": "delete-button" }).props.onClick();
  expect(onClick).toHaveBeenCalled();
});

test("should render FilterButton isSelected false", () => {
  component = TestRenderer.create(
    <FilterButton isSelected={false} onClick={() => {}} text="test" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render FilterButton isSelected true", () => {
  component = TestRenderer.create(
    <FilterButton isSelected={true} onClick={() => {}} text="test" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TabHeader up", () => {
  component = TestRenderer.create(
    <TabHeader name="test" order="true" isSortedHeader={true} sens="up" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TabHeader down", () => {
  component = TestRenderer.create(
    <TabHeader name="test" order="true" isSortedHeader={true} sens="down" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("should render TabHeader no order", () => {
  component = TestRenderer.create(
    <TabHeader name="test" order={null} isSortedHeader={false} sens="down" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
