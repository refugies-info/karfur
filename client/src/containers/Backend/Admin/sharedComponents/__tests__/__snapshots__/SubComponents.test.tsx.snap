// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DeleteButton disabled 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: not-allowed;
}

.c0:hover {
  background-color: #828282;
}

<div
  className="c0"
  data-test-id="delete-button"
  disabled={true}
>
  <div
    style={
      Object {
        "marginBottom": "4px",
      }
    }
  >
    <span>
      <i
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </span>
  </div>
</div>
`;

exports[`should render DeleteButton not disabled 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: #f44336;
}

<div
  className="c0"
  data-test-id="delete-button"
  disabled={false}
  onClick={[Function]}
>
  <div
    style={
      Object {
        "marginBottom": "4px",
      }
    }
  >
    <span>
      <i
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"20px\\" height=\\"20px\\" viewBox=\\"0 0 24 24\\" class=\\"eva eva-trash\\" fill=\\"#ffffff\\"><g data-name=\\"Layer 2\\"><g data-name=\\"trash\\"><rect width=\\"24\\" height=\\"24\\" opacity=\\"0\\"/><path d=\\"M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 4.33c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4z\\"/></g></g></svg>",
          }
        }
      />
    </span>
  </div>
</div>
`;

exports[`should render FilterButton isSelected false 1`] = `
.c0 {
  background: #ffffff;
  color: #212121;
  border-radius: 12px;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  padding: 15px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin-right: 8px;
  cursor: pointer;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

<div
  className="c0"
  onClick={[Function]}
>
  test
</div>
`;

exports[`should render FilterButton isSelected true 1`] = `
.c0 {
  background: #212121;
  color: #ffffff;
  border-radius: 12px;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  padding: 15px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin-right: 8px;
  cursor: pointer;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

<div
  className="c0"
  onClick={[Function]}
>
  test
</div>
`;

exports[`should render SeeButton 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: #212121;
}

<div
  className="c0"
>
  <a
    href="test-burl"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      style={
        Object {
          "marginBottom": "4px",
        }
      }
    >
      <span>
        <i
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"20px\\" height=\\"20px\\" viewBox=\\"0 0 24 24\\" class=\\"eva eva-eye\\" fill=\\"#ffffff\\"><g data-name=\\"Layer 2\\"><g data-name=\\"eye\\"><rect width=\\"24\\" height=\\"24\\" opacity=\\"0\\"/><circle cx=\\"12\\" cy=\\"12\\" r=\\"1.5\\"/><path d=\\"M21.87 11.5c-.64-1.11-4.16-6.68-10.14-6.5-5.53.14-8.73 5-9.6 6.5a1 1 0 0 0 0 1c.63 1.09 4 6.5 9.89 6.5h.25c5.53-.14 8.74-5 9.6-6.5a1 1 0 0 0 0-1zm-9.87 4a3.5 3.5 0 1 1 3.5-3.5 3.5 3.5 0 0 1-3.5 3.5z\\"/></g></g></svg>",
            }
          }
        />
      </span>
    </div>
  </a>
</div>
`;

exports[`should render Structure with null sponsor 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #f44336;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#f44336"
  />
  <div
    className="c2"
  >
    Sans structure
  </div>
</div>
`;

exports[`should render Structure with sponsor actif 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #4caf50;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#4caf50"
  />
  <div
    className="c2"
  >
    test
  </div>
</div>
`;

exports[`should render Structure with sponsor diair 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #f44336;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#f44336"
  />
  <div
    className="c2"
  >
    Structure temporaire
  </div>
</div>
`;

exports[`should render Structure with sponsor en attente 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #FF9800;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#FF9800"
  />
  <div
    className="c2"
  >
    test
  </div>
</div>
`;

exports[`should render Structure with sponsor supprime 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #f44336;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#f44336"
  />
  <div
    className="c2"
  >
    test
  </div>
</div>
`;

exports[`should render Structure with sponsor without nom 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
}

.c1 {
  width: 10px;
  height: 10px;
  background-color: #f44336;
  border-radius: 50%;
  margin-right: 10px;
}

.c2 {
  word-break: break-all;
  max-width: 280px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#f44336"
  />
  <div
    className="c2"
  >
    Sans structure
  </div>
</div>
`;

exports[`should render StyledStatus Accepté structure  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #FF9800;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#FF9800"
  disabled={false}
>
  Accepté
</div>
`;

exports[`should render StyledStatus Brouillon  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #FFEB3B;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #212121;
}

<div
  className="c0"
  color="#FFEB3B"
  disabled={false}
>
  Brouillon
</div>
`;

exports[`should render StyledStatus En attente admin  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #8BC34A;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#8BC34A"
  disabled={false}
>
  À valider
</div>
`;

exports[`should render StyledStatus En attente non prioritaire  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #F44336;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#F44336"
  disabled={false}
>
  Sans structure
</div>
`;

exports[`should render StyledStatus Rejeté structure  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #F44336;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#F44336"
  disabled={false}
>
  Rejeté
</div>
`;

exports[`should render StyledStatus Supprimé  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #F44336;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#F44336"
  disabled={false}
>
  Supprimé
</div>
`;

exports[`should render StyledStatus actif  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #4CAF50;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#4CAF50"
  disabled={false}
>
  Publié
</div>
`;

exports[`should render StyledStatus en attente  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #FF9800;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#FF9800"
  disabled={false}
>
  En attente
</div>
`;

exports[`should render StyledStatus no corresponding status  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #0421b1;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#0421b1"
  disabled={false}
>
  Nouveau !
</div>
`;

exports[`should render StyledStatus no corresponding with color  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: color;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: textColor;
}

<div
  className="c0"
  color="color"
  disabled={false}
>
  test
</div>
`;

exports[`should render StyledStatus no corresponding with override  1`] = `
.c0 {
  font-weight: bold;
  border-radius: 6px;
  padding: 8px;
  background-color: #828282;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  white-space: nowrap;
  cursor: pointer;
  color: #ffffff;
}

<div
  className="c0"
  color="#828282"
  disabled={false}
>
  test
</div>
`;

exports[`should render TabHeader down 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-weight: bold;
  cursor: pointer;
}

<div
  className="c0"
  order={1}
>
  test
  <span>
    <i
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"20px\\" height=\\"20px\\" viewBox=\\"0 0 24 24\\" class=\\"eva eva-chevron-down\\" fill=\\"#212121\\"><g data-name=\\"Layer 2\\"><g data-name=\\"chevron-down\\"><rect width=\\"24\\" height=\\"24\\" opacity=\\"0\\"/><path d=\\"M12 15.5a1 1 0 0 1-.71-.29l-4-4a1 1 0 1 1 1.42-1.42L12 13.1l3.3-3.18a1 1 0 1 1 1.38 1.44l-4 3.86a1 1 0 0 1-.68.28z\\"/></g></g></svg>",
        }
      }
    />
  </span>
</div>
`;

exports[`should render TabHeader no order 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-weight: bold;
  cursor: inherit;
}

<div
  className="c0"
  order={0}
>
  test
</div>
`;

exports[`should render TabHeader up 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-weight: bold;
  cursor: pointer;
}

<div
  className="c0"
  order={1}
>
  test
  <span>
    <i
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </span>
</div>
`;

exports[`should render Title with titre marque 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-style: normal;
  font-size: 16px;
  line-height: 20px;
  max-width: 370px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
}

<div
  className="c0"
>
  <b>
    titre informatif
  </b>
  <span>
    avec titre marque
  </span>
</div>
`;

exports[`should render Title without titre marque 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-style: normal;
  font-size: 16px;
  line-height: 20px;
  max-width: 370px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
}

<div
  className="c0"
>
  <b>
    titre informatif
  </b>
</div>
`;

exports[`should render TypeContenu with demarche, detailed vue 1`] = `
.c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #ffffff;
  background-color: #212121;
  padding: 8px;
  border-radius: 6px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

<div
  className="c0"
>
  Démarche
</div>
`;

exports[`should render TypeContenu with demarche, not detailed vue 1`] = `
.c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #212121;
  background-color: #ffffff;
  padding: 8px;
  border-radius: 6px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

<div
  className="c0"
>
  Démarche
</div>
`;

exports[`should render TypeContenu with dispositif, detailed vue 1`] = `
.c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #ffffff;
  background-color: #212121;
  padding: 8px;
  border-radius: 6px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

<div
  className="c0"
>
  Dispositif
</div>
`;

exports[`should render TypeContenu with dispositif, not detailed vue 1`] = `
.c0 {
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #ffffff;
  background-color: #212121;
  padding: 8px;
  border-radius: 6px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

<div
  className="c0"
>
  Dispositif
</div>
`;

exports[`should render ValidateButton disabled 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: not-allowed;
}

.c0:hover {
  background-color: #828282;
}

<div
  className="c0"
  data-test-id="validate-button"
  disabled={true}
  onClick={[Function]}
>
  <div
    style={
      Object {
        "marginBottom": "4px",
      }
    }
  >
    <span>
      <i
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </span>
  </div>
</div>
`;

exports[`should render ValidateButton not disabled 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  background-color: #828282;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 4px;
  margin-left: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: #4caf50;
}

<div
  className="c0"
  data-test-id="validate-button"
  disabled={false}
  onClick={[Function]}
>
  <div
    style={
      Object {
        "marginBottom": "4px",
      }
    }
  >
    <span>
      <i
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"20px\\" height=\\"20px\\" viewBox=\\"0 0 24 24\\" class=\\"eva eva-checkmark-outline\\" fill=\\"#ffffff\\"><g data-name=\\"Layer 2\\"><g data-name=\\"checkmark\\"><rect width=\\"24\\" height=\\"24\\" opacity=\\"0\\"/><path d=\\"M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z\\"/></g></g></svg>",
          }
        }
      />
    </span>
  </div>
</div>
`;
